File: FE/src/features/company/services/companyService.ts
import { apiClient } from '../../../services/apiClient'
import type { PaginatedResult } from '../../../types/api.types'
import type {
  CompanyResponse,
  CreateCompanyRequest,
  UpdateCompanyRequest,
} from '../types/company.types'

export const companyService = {
  getAll: (page = 1, size = 10, sort = 'id,asc', active?: boolean) => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
      sort,
    })
    if (active !== undefined) {
      params.set('active', String(active))
    }
    return apiClient<PaginatedResult<CompanyResponse>>(`/companies?${params}`)
  },

  getById: (id: number) => apiClient<CompanyResponse>(`/companies/${id}`),

  create: (data: CreateCompanyRequest) =>
    apiClient<CompanyResponse>('/companies', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  update: (data: UpdateCompanyRequest) =>
    apiClient<CompanyResponse>('/companies', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  disable: (id: number) =>
    apiClient<CompanyResponse>(`/companies/${id}`, { method: 'DELETE' }),

  enable: (id: number) =>
    apiClient<CompanyResponse>(`/companies/${id}/enable`, { method: 'POST' }),
}

File: BE/src/main/java/Manager_vnd/Manager/feature/company/CompanyService.java
package Manager_vnd.Manager.feature.company;

import java.util.List;

import Manager_vnd.Manager.dto.PaginatedResult;
import Manager_vnd.Manager.feature.company.dto.CompanyResponse;
import Manager_vnd.Manager.feature.company.dto.CreateCompanyRequest;
import Manager_vnd.Manager.feature.company.dto.PublicCompanyResponse;
import Manager_vnd.Manager.feature.company.dto.UpdateCompanyRequest;

public interface CompanyService {

    List<PublicCompanyResponse> listPublic();

    PaginatedResult<CompanyResponse> getAll(int page, int size, String sort, Boolean active);

    CompanyResponse getById(long id);

    CompanyResponse create(CreateCompanyRequest request);

    CompanyResponse update(UpdateCompanyRequest request);

    CompanyResponse disable(long id);

    CompanyResponse enable(long id);
}
